polkadot/xcm/xcm-executor/src/traits/asset_exchange.rs

```diff
@@ -37,6 +37,15 @@ pub trait AssetExchange {
 		want: &Assets,
 		maximal: bool,
 	) -> Result<AssetsInHolding, AssetsInHolding>;
+
+	/// Handler for quoting the exchange price of two assets.
+	///
+	/// - `asset1` The first asset.
+	/// - `asset2` The second asset.
+	/// - `maximal`: If `true`, then all of `asset1` should be used.
+	fn quote_exchange_price(_asset1: &Asset, _asset2: &Asset, _maximal: bool) -> Option<u128> {
```

Comment:
The doc and signature does not set a behaviour for this quote. I get a price for both assets? 
We probably mean quote the amount of asset1 to get at least asset2 amount.

Why not something like, for symmetry? Do we plan to add one more function when we need it for `exchange_asset`?
`fn quote_exchange_price(give: &Assets, want: &Assets, maximal: bool) -> Option<Assets>`

Reply:
Symmetry between the two functions was discussed, see https://github.com/paritytech/polkadot-sdk/pull/5130#discussion_r1692792236

I should improve the doc accordingly

Reply:
Yes, still not clear why not use collections. The conclusion in that thread that a given API serves our specific case, this is why it's good enough, when we need collection, we break that API.
Basically we introducing this function based on our use case (for single exchange adapter), and not based on the existing asset exchange contract (AssetExchange trait).

Reply:
I'll change it to using collections then